src/apps/ipv4/arp.lua

```diff
@@ -191,6 +195,11 @@ function ARP:arp_resolved (ip, mac, provenance)
       resolve_alarm:clear()
    end
    self.next_mac = mac
+   if self.next_mac then
+      local buf = ffi.new('union { uint64_t u64; uint8_t bytes[6]; }')
```

Comment:
I wonder if it makes sense to change the definition here: https://github.com/Igalia/snabb/blob/lwaftr/src/lib/protocol/ethernet.lua#L12 and use that instead of defining it twice here and in the NDP code. WDYT?

Reply:
I don't know, maybe. Definitely it would be convenient, but at first glance to me it depends much on other people's needs for converting from u64 to MAC string and viceversa. So far we're the only users so I would prefer not to change `protocol/ethernet.lua`. However this very same code is defined at https://github.com/Igalia/snabb/blob/lwaftr/src/program/top/top.lua#L511 so at some point I think it'd be worth put it somewhere in an util library (or do the change you suggested).